### PR TITLE
refactor(flow): Delete wallet flow refactored

### PR DESCRIPTION
**Pull request by amanCypherock**

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4828 Task controller merge - Delete wallet flow](https://app.clickup.com/t/9002019994/PRF-4828)

**Comment by amanCypherock** [comment]
Converted to draft because partial wallet case is not handled yet.

File: src/wallet/delete_wallet_flow.c [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  ASSERT(NULL != wallet_id);
+
+  if (!core_scroll_page(
+          NULL, ui_text_need_all_x1cards_to_delete_wallet_entirely, NULL)) {
+    return;
+  }
+
+  memzero(&wallet, sizeof(wallet));
+  memzero(&wallet_shamir_data, sizeof(wallet_shamir_data));
+  memzero(&wallet_credential_data, sizeof(wallet_credential_data));
```
Comment (kartikCypherock) [inline src/wallet/delete_wallet_flow.c:143]: Use common function implemented in [view seed flow](https://github.com/Cypherock/x1_wallet_firmware/pull/339)

File: src/wallet/delete_wallet_flow.c [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+extern Wallet_credential_data wallet_credential_data;
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+static bool get_pin_input() {
+  if (!WALLET_IS_PIN_SET(wallet.wallet_info)) {
+    return true;
+  }
+
+  memzero(&wallet_credential_data, sizeof(wallet_credential_data));
```
Comment (kartikCypherock) [inline src/wallet/delete_wallet_flow.c:110]: Clear only `password_single_hash` field?

File: src/wallet/delete_wallet_flow.c [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  memzero(&wallet_shamir_data, sizeof(wallet_shamir_data));
+  memzero(&wallet_credential_data, sizeof(wallet_credential_data));
+
+  if (false == get_wallet_data_by_id(wallet_id, &wallet)) {
+    mark_core_error_screen(ui_text_something_went_wrong);
+    return;
+  }
+
+  while (1) {
+    if (!get_pin_input()) {
```
Comment (kartikCypherock) [inline src/wallet/delete_wallet_flow.c:151]: Check if having state based approach will help in readability?

File: src/wallet/delete_wallet_flow.c (hunk 1 above)
Comment (amanCypherock) [reply src/wallet/delete_wallet_flow.c]: Added, was waiting for merges

File: src/wallet/delete_wallet_flow.c (hunk 2 above)
Comment (amanCypherock) [reply src/wallet/delete_wallet_flow.c:110]: fixed, double hash field also memzeroed

File: src/wallet/delete_wallet_flow.c (hunk 3 above)
Comment (amanCypherock) [reply src/wallet/delete_wallet_flow.c:151]: Considering there are only two states, it doesn't make sense to fix.

**Review by kartikCypherock — APPROVED** [review]

**Pull request merged**